logrotate/Options.cmake

```diff
@@ -9,27 +9,50 @@
 #
 # example: cmake -DBUILD_STATIC=ON .. 
 option (BUILD_STATIC "Building g3logrotate as static library. Linking with static g3log library" OFF)
-IF (BUILD_STATIC) 
+IF (BUILD_STATIC)
   SET(G3LOG_LIBRARY_TYPE g3logger)
-  SET(G3LOG_LIBRARY_NAME libg3logger.a)
   MESSAGE("'OPTION: cmake -DBUILD_STATIC=ON'\nLinking statically with g3log. Creating static library for g3logrotate")
+  
+  IF (MSVC)
+    SET(G3LOG_LIBRARY_NAME_DEBUG g3logger_d.lib)
+	SET(G3LOG_LIBRARY_NAME_RELEASE g3logger.lib)
```

Comment:
That is correct but msvc supports multi-configuration so both are generated
at the same time.
So for example if you want to generate g3sink debugs you will have to link
to the debug version as if you want release it will be the release version.

Since both are generated by cmake configure both have to be defined.
It is exactly the same things with Boost. Why do you need both debug and
release defined in cmake ?
Similarly you need both here.

On 19 December 2015 at 00:41, Kjell Hedström <notifications@github.com>
wrote:

> In logrotate/Options.cmake
> <https://github.com/KjellKod/g3sinks/pull/20#discussion_r48089214>:
>
> >    MESSAGE("'OPTION: cmake -DBUILD_STATIC=ON'\nLinking statically with g3log. Creating static library for g3logrotate")
> > +
> > +  IF (MSVC)
> > +    SET(G3LOG_LIBRARY_NAME_DEBUG g3logger_d.lib)
> > +	SET(G3LOG_LIBRARY_NAME_RELEASE g3logger.lib)
>
> I am curious about your reasoning for this.
>
> Why would you need two paths for the g3sinks library? Either you are
> compiling it for debug or you are compiling it for release, right? You will
> never compile it for both release and debug in the same go.
>
> What do you gain for adding G3_LIBRARY_PATH_RELEASE instead
> G3_LIBRARY_PATH and using the standard Cmake way of defining debug or
> release builds?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/KjellKod/g3sinks/pull/20/files#r48089214>.
>